discord/app.py

```diff
@@ -178,6 +178,18 @@ def remove_option(self, name: str) -> AutoCompleteResponseT:
 
     def __iter__(self):
         return iter([{"name": k, "value": v} for k, v in self.items()])
+        
+    def __getitem__(self, x: Union[slice, int]) -> AutoCompleteResponseT:
+        if isinstance(x, slice):
+            self = AutoCompleteResponse({
+                key: value
+                for key, value in list(self.items())[x]
```

Comment:
You're doing a double iteration here, might be faster just to iterate once with a for loop